app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,52 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: [:show, :edit, :update, :destroy]
```

Comment:
This is fine, and I like that you're able to do it if you want to, but it always seems like more trouble than it's worth to me. It feels like I'm hiding code where I could just be including it in the particular controller action and making it plain what was happening there.